mypy/checkstrformat.py

```diff
@@ -49,21 +50,24 @@ class StringFormatterChecker:
     def __init__(self,
                  exprchk: 'mypy.checkexpr.ExpressionChecker',
                  chk: 'mypy.checker.TypeChecker',
-                 msg: MessageBuilder) -> None:
+                 msg: MessageBuilder,
+                 pyversion: Tuple[int, int]) -> None:
         """Construct an expression type checker."""
         self.chk = chk
         self.exprchk = exprchk
         self.msg = msg
+        self.pyversion = pyversion
 
     # TODO: In Python 3, the bytes formatting has a more restricted set of options
     # compared to string formatting.
-    # TODO: Bytes formatting in Python 3 is only supported in 3.5 and up.
     def check_str_interpolation(self,
                                 str: Union[StrExpr, BytesExpr, UnicodeExpr],
                                 replacements: Node) -> Type:
         """Check the types of the 'replacements' in a string interpolation
         expression: str % replacements
         """
+        if isinstance(str, BytesExpr) and self.pyversion[0] == 3 and self.pyversion[1] < 5:
+            assert False
```

Comment:
I don't think this is the right approach. It's probably better to move the version check to whatever calls check_str_interpolation(), so that it doesn't even get called for bytes if pyversion < 3.5, and then the fallback logic in the caller (which just maps '%' to a call to `__mod__`) will take over.